client/jetpack-connect/plans.jsx

```diff
@@ -210,6 +211,10 @@ class Plans extends Component {
 		// clears whatever we had stored in local cache
 		this.props.selectPlanInAdvance( null, this.props.selectedSiteSlug );
 
+		if ( cartItem.product_slug === get( this.props, 'selectedSite.plan.product_slug', null ) ) {
```

Comment:
Wouldn't this break if there's nothing in the cart? This may also work in unexpected ways if we start preselecting the free plan, as suggested in #15741.

I'd suggest that we move this check below the `if ( ! cartItem || cartItem.product_slug === PLAN_JETPACK_FREE ) {` one just in case, unless we have a specific reason not to.